### PR TITLE
Upgrade to new phosphor mono repo.

### PR DESCRIPTION
**Pull request by afshin**
### Notes

- Because the `MainMenu` API has changed and cannot just simply `addItem`, it instead offers to add a menu (`addMenu`). The plugin will need a `getMenu` option that returns a menu if it exists so that user code will be able to add items.
- Because the application shell is new (it does not come from `phosphide` any more) and because the old `SideBar` class is gone and now we just use a `TabBar` (`phosphor/lib/ui/tabbar`), we need new CSS for the left area.
- The horizontal `TabBar` CSS also needs revision.
- The menu CSS needs revision.
- Tests need to be ported to the mono-repo.
- Examples need to be ported to the mono-repo.

### CODE REVIEW

File: src/application/shell.ts [hunk 1]
```diff
@@ -0,0 +1,389 @@
...
+    this._leftHandler = leftHandler;
+    this._rightHandler = rightHandler;
+
+    topPanel.id = 'jp-top-panel';
+    hsplitPanel.id = 'jp-main-split-panel';
+    leftHandler.sideBar.addClass('jp-mod-left');
+    rightHandler.sideBar.addClass('jp-mod-right');
+    leftHandler.stackedPanel.id = 'jp-left-stack';
+    rightHandler.stackedPanel.id = 'jp-right-stack';
+    dockPanel.id = 'p-main-dock-panel';
```
Comment (blink1073) [inline src/application/shell.ts]: jp-

File: src/mainmenu/plugin.ts [hunk 2]
```diff
@@ -70,43 +89,31 @@ namespace MainMenu {
...
+  activate: activateMainMenu
 };
 
 
 /**
  * Activate the main menu extension.
  */
-function activateMainMenu(app: Application): void {
+function activateMainMenu(lab: JupyterLab): MainMenu {
+  Private.menuBar = new MenuBar({ keymap: lab.keymap });
```
Comment (blink1073) [inline src/mainmenu/plugin.ts]: No need for the module level state.

Comment (afshin) [inline src/mainmenu/plugin.ts]: I need access to the `menuBar` instance so I can refer to it in `MainMenu#addMenu`.

Comment (blink1073) [inline src/mainmenu/plugin.ts]: My point is there should be no need to do this now that we have merged extensions and services, but we can save refactoring for later.

File: src/docmanager/widgetmanager.ts [hunk 3]
```diff
@@ -88,15 +88,15 @@ class DocumentWidgetManager {
...
+   * Install the message hook for the widget and add to list
    * of known widgets.
    */
   adoptWidget(id: string, widget: Widget): void {
     if (!(id in this._widgets)) {
       this._widgets[id] = [];
     }
     this._widgets[id].push(widget);
-    installMessageFilter(widget, this);
+    installMessageHook(widget, this.filterMessage);
```
Comment (blink1073) [inline src/docmanager/widgetmanager.ts]: Needs to be bound?

File: src/notebook/plugin.ts [hunk 4]
```diff
@@ -102,52 +106,47 @@ const cmdIds = {
 };
 
 
+/* tslint:disable */
 /**
- * A class that tracks notebook widgets.
+ * The clipboard token.
```
Comment (blink1073) [inline src/notebook/plugin.ts]: Wrong docstring

File: src/terminal/plugin.ts [hunk 5]
```diff
@@ -47,136 +52,110 @@ const LANDSCAPE_ICON_CLASS = 'jp-MainAreaLandscapeIcon';
...
-      handler: increaseFont
-    },
-    {
-      id: decreaseTerminalFontSize,
-      handler: decreaseFont
-    },
-    {
-      id: toggleTerminalTheme,
-      handler: toggleTheme
+  commands.addCommand(newTerminalId, {
```
Comment (blink1073) [inline src/terminal/plugin.ts]: Dupilciated command

File: src/common/observablelist.ts [hunk 6]
```diff
@@ -628,14 +628,17 @@ class ObservableList<T> implements IObservableList<T> {
...
-    while (items.length) {
-      this.internal.insert(i++, items.shift());
+    let i = index;
+    let j = 0;
+    let len = items.length;
+    while (j < len) {
+      this.internal.insert(i++, items[j++]);
     }
-
+    console.log('items', items);
```
Comment (blink1073) [inline src/common/observablelist.ts]: Debug statements

**Comment by blink1073** [comment]
:tada:

**Pull request merged**